bookwyrm/templates/book/edit_book.html

```diff
@@ -4,6 +4,45 @@
 
 {% block title %}{% if book %}{% blocktrans with book_title=book.title %}Edit "{{ book_title }}"{% endblocktrans %}{% else %}{% trans "Add Book" %}{% endif %}{% endblock %}
 
+{% block scripts %}
+<script defer>
+ document.getElementById("id_first_published_year_only")
+         .addEventListener(
+             "input",
+             setToYearOnly(
+                 "id_first_published_date",
+                 "label_first_published_date",
+                 "label_first_published_year_only"),
+             {passive: true});
+ document.getElementById("id_published_year_only")
+         .addEventListener(
+             "input",
+             setToYearOnly(
+                 "id_published_date",
+                 "label_published_date",
+                 "label_published_year_only"),
+             {passive: true});
+ function setToYearOnly(date_input_id, date_label_id, year_label_id) {
+     function setFirstPublishedToYearOnly(event) {
+         let first_published_date_input = document.getElementById(date_input_id);
+         let first_published_date_label = document.getElementById(date_label_id);
+         let first_published_year_label = document.getElementById(year_label_id);
+         // potential HACK: older ie browsers require a different method call: https://gist.github.com/tzi/2953155#file-important-js-L5
+         if (event.target.checked) {
+             first_published_date_label.setAttribute("style", "display: none !important");
+             first_published_year_label.setAttribute("style", "display: block");
+             first_published_date_input.type = "number";
+         } else {
+             first_published_date_label.setAttribute("style", "display: block");
+             first_published_year_label.setAttribute("style", "display: none !important");
+             first_published_date_input.type = "date";
+         }
```

Comment:
Your comment raises an interesting question: Does BookWyrm want to define a set of supported browser for a full-feature experience?

If yes, which ones? (That would definitely go into a [.browserslistrc](https://github.com/browserslist/browserslist) file. :) )

Reply:
my personal projects are always "the browsers supported are whatever browsers support the features i need without polyfills and work-arounds". that's definitely not accessible enough for a project with more reach and dev power though (imo)

i recently read an article (that i've lost, sadly) talking about how few people *actually* use internet explorer. my personal opinion is that ie is end-of-life for several years now, bookwyrm's potential userbase probably has no overlap with *any* of the people using ie, and the added size of polyfills and work-arounds is more of an accessibility issue than one or two people being unable to use certain js features

as for non-ie browsers to be supported, i'd have to defer to you! i think the js here makes it pretty clear that i'm not a front-end person :p

Reply:
I completely second the progressive enhancement approach.

My question was more about a **full**-feature experience, where **feature** meant UX/UI features more than the content provided. I hope the intention is clearer. :)

I won’t regret not caring about IE **specifically** as long as the content is available to the few people using it.

Reply:
i definitely understand what you mean now! i think it might be a good idea to explicitly list several target browsers, so we can make sure we're usable with more than just the main two or three